src/keys/PasswordKey.h

```diff
@@ -30,13 +30,16 @@ class PasswordKey : public Key
 
     PasswordKey();
     explicit PasswordKey(const QString& password);
+    ~PasswordKey() override;
     QByteArray rawKey() const override;
     void setPassword(const QString& password);
 
     static QSharedPointer<PasswordKey> fromRawKey(const QByteArray& rawKey);
 
 private:
-    QByteArray m_key;
+    static constexpr int SHA256_SIZE = 32;
+
+    char* m_key = nullptr;
```

Comment:
Shouldn't m_key be defined in ```Key```? It is used in all keys.

Reply:
Yes and no. I kinda want to keep it as an implementation detail. Allocation and deallocation should be in the same class and it wouldn't make a lot of sense to move it to the parent class for any other reason. Also, other keys might just be a wrapper for an external key facility that don't need an internal buffer at all.